Tests/test_SeqIO_QualityIO.py

```diff
@@ -1057,6 +1057,7 @@ def failure_check(self, filename, in_format, out_format):
         err2 = str(cm.exception)
         # Verify that parse/write and convert give the same failure
         err_msg = f"{msg}: parse/write and convert gave different failures"
+        # TODO: Do we really want to enforce this?
```

Comment:
It would be nice, the convert code branch is meant to be an optimized version of the parse into SeqRecord and back to disk. Perhaps it wouldn't be too bad to add your improved error messages there?

Reply:
OK, they should be the same now